src/methods/query/getPoolInfo.ts

```diff
@@ -36,7 +44,62 @@ export async function getPoolInfo() {
           poolAssetNames: [],
         };
       });
+
+      for (let i = 0; i < allPools.length; i++) {
+        const [key, value] = allPools[i];
+
+        const poolId = key.toHuman()?.toString() || 'ERR';
+        const poolAssets = api
+          .createType('Vec<u32>', value)
+          .map(assetId => assetId.toNumber())
+          .sort((a, b) => a - b);
+        let promises = parsedPoolsList.map(poolInfo => {
+          const assetPair = poolInfo[1];
+  
+          if (assetPair && Array.isArray(assetPair) && assetPair[0] && assetPair[1]) {
+            return getPoolAssetsAmounts(assetPair[0].toString(), assetPair[1].toString());
+          }
+          
+          return null;
+        });
   
+        promises = promises.filter(promise => promise);
+  
+        let poolAssetAmounts: any[] = await Promise.all(promises);
+
+        let promises2 = poolAssetAmounts.map((assetAmounts) => {
+          if (assetAmounts) {
+            return wasm.get_spot_price(assetAmounts.asset1, assetAmounts.asset2, '1000000000000');
+          }
+          return null;
+        });
+  
+        promises2 = promises2.filter(promise => promise);
+  
+        let spotPrices = await Promise.all(promises2);
+  
+        for (let i = 0; i < spotPrices.length; i++) {
+          poolAssetAmounts[i].spotPrice = new BigNumber(spotPrices[i]);
+        }
+
+        const poolAssetsAmount = await getPoolAssetsAmounts(poolAssets[0].toString(), poolAssets[1].toString());
+        const assetPrices = getAssetPrices(tokenTradeMap, parsedPoolsList, poolAssetAmounts);
+
+        if (poolAssetsAmount && poolAssetsAmount.asset1 && poolAssetsAmount.asset2) {
+          const asset1Price = (new BigNumber(poolAssetsAmount.asset1)).multipliedBy(assetPrices[poolAssets[0]]).dividedBy(priceUnit);
+          const asset2Price = (new BigNumber(poolAssetsAmount.asset2)).multipliedBy(assetPrices[poolAssets[1]]).dividedBy(priceUnit);
+          const marketCap = asset1Price.plus(asset2Price);
+
+          if (poolInfo[poolId]) {
+            poolInfo[poolId].poolAssetsAmount = {
+              asset1: poolAssetsAmount.asset1,
+              asset2: poolAssetsAmount.asset2,
```

Comment:
I guess, asset1 and asset2 values should be in BigNumber but not in string. 
Also results of `getPoolInfo` functions should be converted with `toExternalBN ` for final user, like other BigNumber values.